tests-api/spec/use_cases/calculate_score_spec.rb

```diff
@@ -126,11 +126,37 @@
         results: results,
       }
     end
-    let(:score) { "#{answers.size}/#{answers.size}" }
-    let(:results) do
-      test.questions.map { |q| { question_uuid: q.id, correct: true } }
+
```

Comment:
how about something like this to remove duplication?
```ruby
describe "#data" do
  subject(:data) { described_class.call(params: params).data }

  # params
  let(:answers) do
    [{ question_uuid: question_uuid, selected_choices: selected_choices }]
  end

  # data
  let(:expected_data) do
    { score: score, results: results }
  end
  let(:results) do
    [{ question_uuid: question_uuid, correct_answers: correct_answers, correct: correct }]
  end

  let(:question) { test.questions.first }
  let(:question_uuid) { question.id }

  let(:correct_answers) do
    question.choices.select { |c| c[:correct] }.map { |c| c[:answer] }
  end

  describe "when results includes correct submitted answers" do
    let(:score) { "1/1" }
    let(:selected_choices) { correct_answers }
    let(:correct) { true }
    # ...

  describe "when results includes incorrect submitted answers" do
    let(:score) { "0/1" }
    let(:selected_choices) { ["Incorrect answer"] }
    let(:correct) { false }
    # ...
```

Reply:
Added some of such changes.